core/test/base/batch_lin_op.cpp

```diff
@@ -0,0 +1,405 @@
+{
+    op->apply(alpha, b, beta, x);
+
+    ASSERT_EQ(op->last_access, ref2);
+}
+
+
+TEST_F(EnableBatchLinOp, CallsExtendedApplyImplBatch)
+{
+    op2->apply(alpha2, b2, beta2, x2);
+
+    ASSERT_EQ(op2->last_access, ref2);
+}
+
+
+TEST_F(EnableBatchLinOp, ApplyFailsOnWrongBatchSize)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{3, 4}));
+
+    ASSERT_THROW(op->apply(wrong, x), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ApplyFailsOnWrongNumBatchItems)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{3, 4}));
+
+    ASSERT_THROW(op2->apply(wrong, x2), gko::ValueMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ApplyFailsOnWrongSolutionRows)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{5, 4}));
+
+    ASSERT_THROW(op->apply(b, wrong), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ApplyFailsOnOneBatchItemWrongSolutionRows)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(2, gko::dim<2>{5, 4}));
+
+    ASSERT_THROW(op2->apply(b2, wrong), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ApplyFailsOnWrongSolutionColumns)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{3, 5}));
+
+    ASSERT_THROW(op->apply(b, wrong), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ApplyFailsOnOneBatchItemWrongSolutionColumn)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(2, gko::dim<2>{3, 5}));
+
+    ASSERT_THROW(op2->apply(b2, wrong), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ExtendedApplyFailsOnWrongBatchSize)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{3, 4}));
+
+    ASSERT_THROW(op->apply(alpha, wrong, beta, x), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ExtendedApplyFailsOnWrongSolutionRows)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{5, 4}));
+
+    ASSERT_THROW(op->apply(alpha, b, beta, wrong), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ExtendedApplyFailsOnWrongSolutionColumns)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{3, 5}));
+
+    ASSERT_THROW(op->apply(alpha, b, beta, wrong), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ExtendedApplyFailsOnWrongAlphaDimension)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{2, 5}));
+
+    ASSERT_THROW(op->apply(wrong, b, beta, x), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ExtendedApplyFailsOnWrongBetaDimension)
+{
+    auto wrong =
+        DummyBatchLinOp::create(ref, gko::batch_dim<2>(1, gko::dim<2>{2, 5}));
+
+    ASSERT_THROW(op->apply(alpha, b, wrong, x), gko::DimensionMismatch);
+}
+
+
+TEST_F(EnableBatchLinOp, ApplyDoesNotCopyBetweenSameMemory)
```

Comment:
I think this and the following test are better done by using a logger to check if a copy event was issued.

Reply:
I think these tests are not relevant as they mainly test behaviour of `make_temporary_clone`, so I have removed them.